src/chainstate/burn/db/sortdb.rs

```diff
@@ -668,6 +675,19 @@ impl <'a> SortitionHandleConn <'a> {
         let chain_tip = SortitionId(chain_tip.0.clone());
         SortitionHandleConn::open_reader(connection, &chain_tip)
     }
+    
+    /// open a reader handle from a consensus hash
+    pub fn open_reader_consensus_stubbed(connection: &'a SortitionDBConn<'a>, chain_tip: &ConsensusHash) -> Result<SortitionHandleConn<'a>, db_error> {
+        let sn = match SortitionDB::get_block_snapshot_consensus(&connection.conn, chain_tip)? {
+            Some(sn) => sn,
+            None => {
+                test_debug!("No such chain tip consensus hash {}", chain_tip);
+                return Err(db_error::NotFoundError);
+            }
+        };
+
+        SortitionHandleConn::open_reader_stubbed(connection, &sn.burn_header_hash)
```

Comment:
This shouldn't need to be marked stubbed -- a reader could be opened with `&sn.sortition_id`.

Reply:
Fixed